File: python/section1/q06.py
from q05 import make_ngram
string1 = "paraparaparadise"
string2 = "paragraph"
X = set(''.join(gram) for gram in make_ngram(sequence=string1))
Y = set(''.join(gram) for gram in make_ngram(sequence=string2))
uniXY = X.union(Y)
intXY = X.intersection(Y)
diffXY = X.difference(Y)
X_contains_se = X.__contains__('se')
Y_contains_se = Y.__contains__('se')

print('X:', X)
print('Y:', Y)
print('X ∪ Y:', uniXY)
print('X ∩ Y:', intXY)
print('X - Y:', diffXY)
print('X contains se:', X_contains_se)
print('Y contains se:', Y_contains_se)

File: scala/section1/src/q06.scala
/**
  * Created by nomotoeriko on 2017/04/20.
  */
object q06 {
  def main(args: Array[String]): Unit = {
    val sentence1 = "paraparaparadise"
    val sentence2 = "paragraph"
    val X = q05.make_ngram(2, sentence1.toList).toSet
    val Y = q05.make_ngram(2, sentence2.toList).toSet
    println("X ∪ Y: %s".format(X|Y))
    println("X ∩ Y: %s".format(X&Y))
    println("X - Y: %s".format(X.diff(Y)))
    println("X contains se: %s".format(X("se".toList)))
    println("Y contains se: %s".format(Y("se".toList)))
  }
}
